homework/src/app.js

```diff
@@ -2,10 +2,192 @@
+     "0" == [] // it is false because there is a truthy and falsy conflict at this equality. 
+
+// 2. 
+const x = [1, 2, 3];
+const y = [1, 2, 3];
+const z = y;
+
+    // x == y,   it is false because x and y reference to different values. 
+    // x === y,  it is false because x and y reference to different values.
+    // z === y,  this is true since they reference to same value.
+    // z == x ,  it is false because x and y reference to different values.
+
+// 3. 
+const o1 = { foo: "bar" };
+const o2 = { foo: "bar" };
+const o3 = o2;
+
+// Show that changing o2 changes o3 (or not) and changing o1 changes o3(or not).
+    // Changing o2 changes o3 since o3 is referenced to value of o2.
+    // But changing o1 doesn`t change o3 since they are not correlated.
+
+//Does the order that you assign (o3 = o2 or o2 = o3) matter?
+    // It does matter since o2 is a constant and cannot be reassigned.
   /* QUESTIONS FROM HYF REPOS */
+
+// 8.1 First declare at least four variables and assign them different data types.
+
+let x = '123';
+
+let y = 123;
+
+let arr = ['John'];
+
+let boo = true;
+
+// 8.2 For each variable write a console.log statement that logs the value
+
+console.log("The value of my variable x is: +x");
+
+console.log("The value of my variable y is: +y");
+
+console.log("The value of my variable arr is: +arr");
+
+console.log("The value of my variable boo is: +boo");
+
+
+
+// 8.3 Now write a console.log statement wherein you first explain in words what you think the type of your variables is.
+
+console.log("The type of my variable x is: string");
+
+console.log("The type of my variable y is: number");
+
+console.log("The type of my variable arr is: object");
+
+console.log("The type of my variable boo is: boolean");
+
+// 8.4 Now use typeof to log the actual type of your variables.
+
+console.log(typeof x);
+
+console.log(typeof y);
+
+console.log(typeof arr);
+
+console.log(typeof boo);
+
+// 8.5 Now compare the types of your different variables with one another.
+
+console.log(typeof x === typeof y); 
+
+console.log(typeof x === typeof arr);
+
+console.log(typeof x === typeof boo);
+
+console.log(typeof y === typeof arr);
+
+console.log(typeof y === typeof boo);
+
+console.log(typeof arr === typeof boo);
+
+
+//8.6 Make sure to also show a message when the variables you are comparing are not the same type.
+
+if (typeof x === typeof y) {
+    console.log('SAME TYPE');
+  } else {
+    console.log('NOT SAME TYPE');
+}
+
+
+
+// 9. If x equals 7, and the only other statement is x = x % 3, what would be the new value of x?
+
+let x = 7;
+x = x % 3 ;
+console.log('The new value of x is : 1 ');
+console.log(x);
+
+// 9.1 Add at least 3 console.log statements in which you show that you understand what % does.
+
+let a = 3;
+a = a % 3; // 0
+console.log(a);
+
+let b = 5;
+b = b % 5; // 0
+console.log(b);
+
+let c = a * b;
+c = c % 15; // 0
+console.log(c);
+
+let d = c + 2;
+d= d % 15; // 2
+console.log(d);
+
```

Comment:
last one is not correct (0+2 % 15 = 1). Furthermore using same variable as assignment variable and operand as in b = b % 5 will work in JavaScript but you are using the same variable for 2 purposes - hold a value and hold a new value I would do let d = c % 15.